src/utils/src/arrow-utils.ts

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {Table as ArrowTable, Field as ArrowField, ListVector, FloatVector} from 'apache-arrow';
+import {DATA_TYPES as AnalyzerDATA_TYPES} from 'type-analyzer';
+import Console from 'global/console';
+import normalize from '@mapbox/geojson-normalize';
+import {parseSync} from '@loaders.gl/core';
+import {WKBLoader, WKTLoader} from '@loaders.gl/wkt';
+import {binaryToGeometry} from '@loaders.gl/gis';
+import {
+  ALL_FIELD_TYPES,
+  ARROW_GEO_METADATA_KEY,
+  ARROW_GEO_METADATA_VERSION
+} from '@kepler.gl/constants';
+import {ProcessorResult, Field} from '@kepler.gl/types';
+import {
+  Feature,
+  MultiPolygon,
+  Position,
+  Polygon,
+  MultiPoint,
+  Point,
+  MultiLineString,
+  LineString
+} from 'geojson';
+
+import {analyzerTypeToFieldType, arrowDataTypeToAnalyzerDataType} from './dataset-utils';
+
+export type RawArrowFeature = {
+  encoding?: string;
+  data: any;
+};
+
+// eslint-disable-next-line no-shadow
+export enum GeoArrowEncodings {
+  MULTI_POLYGON = 'geoarrow.multipolygon',
+  POLYGON = 'geoarrow.polygon',
+  MULTI_LINESTRING = 'geoarrow.multilinestring',
+  LINESTRING = 'geoarrow.linestring',
+  MULTI_POINT = 'geoarrow.multipoint',
+  POINT = 'geoarrow.point',
+  WKB = 'wkb',
+  WKT = 'wkt'
+}
+
+/**
+ * Parse a arrow table with geometry columns and return a dataset
+ *
+ * @param arrowTable the arrow table to parse
+ * @returns dataset containing `fields` and `rows` or null
+ */
+export function processArrowTable(arrowTable: ArrowTable): ProcessorResult | null {
+  if (!arrowTable) {
+    return null;
+  }
+  const metadata = arrowTable.schema.metadata;
+  // get geometry columns if metadata has key 'geo'
+  let geometryColumns = [];
+  if (metadata.get(ARROW_GEO_METADATA_KEY) !== undefined) {
```

Comment:
Note that this is a geoparquet construct, not a geoarrow construct. As of yet, GeoArrow has no table-level metadata. The GeoArrow metadata sets extension metadata on a specific _field_, which you can check by reading `ARROW:extension:name` https://github.com/geoarrow/deck.gl-layers/blob/509547593e50cfd79aa9bc784970679747ea8acc/src/utils.ts#L33-L44

Reply:
Thanks, @kylebarron! Will update by checking `ARROW:extension:name` for each field.

Reply:
Well, anytime you have data that comes from parquet, it will have the `geo` key. So it's not that you _shouldn't_ check the `geo` key, just to be clear that "pure geoarrow" data won't have that